07_train_final_classifiers/scripts/run_all_genes.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+PARAMS_DIR=./02_classify_mutations/results/merged_all_params
+RESULTS_DIR=./07_train_final_classifiers/results/merged_all_params
+ERRORS_DIR=./final_merged_errors
+SEED=42
+
+mkdir -p $ERRORS_DIR
+
+# read the list of cosmic genes from tsv file
+merged_filename="data/cancer_genes/merged_with_annotations.tsv"
+
+read_genes_from_file() {
+    # create global gene array
+    declare -a -g genes
+
+    # read tab-separated file, genes should be the first column
+    while IFS=$'\t' read -r gene class; do
+        genes+=("$gene")
+    done < "$1"
+
+    # remove header
+    genes=("${genes[@]:1}")
+}
+read_genes_from_file $merged_filename
+
+for gene in "${genes[@]}"; do
+
+    # just do expression for now
+    # TODO: could do methylation but would need to pass PCA features
```

Comment:
If you do this in the future (adding methylation), you might be interested in handling several arguments to this script. I was recently looking at ways to handle bash arguments more nicely, and found this code, in case it's helpful in the future: https://stackoverflow.com/a/14203146

Reply:
This is good to know! I think we're going to stick with gene expression for now since the expression models are most relevant/easiest to interpret for most people (gene features vs. CpG features) but I'll keep this in mind.

Reply:
Yes, absolutely, I just suggested that because you have a comment saying `TODO could do methylation...`